docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,285 @@
+ The process generates a list of [search labels](#search-labels) based on skr's physical attributes.
+ The [search labels](#search-labels) are used to query Gardener's Secret Bindings. 
+ [search labels](#search-labels) are generated by evaluating a Rule consisting of Rule Entries and if an entry applies to the request then the rule outputs its specific [search labels](#search-labels) to query the correct HAP Pool.
+ More about HAP process can be found in [Hyperscaler Account Pool](03-10-hyperscaler-account-pool.md) document.
+
+![Rules Evaluation Process](../assets/hap-rules-general.drawio.svg)
+
+## Rule Configuration
+
+The Rule is configured in values.yaml in `hap.rule` property.
+It is is a list of strings, each of which is a single Rule Entry that corresponds to a single HAP Pool. 
+Each Rule Entry is evaluated during a cluster provisioning process.
+If more than one rule applies the best matching rule is selected based on the rule [priority](#priority). The following examples presents the Rule Configuration in values.yaml:
+
+```
+hap: 
+  rule: 
+  - rule_1
+  - rule_2
+  ...
+  - rule_n
+```
+
+Every HAP Pool that the Rule Entry represents must be preconfigured in a Gardener Cluster that KEB connects to.
+
+## Rule Format 
+
+The following example shows the Rule Entry format:
+
+```
+PLAN(INPUT_ATTR_1=VAL_1, INPUT_ATTR_2=VAL_2, ..., INPUT_ATTR_N=VAL_N) -> OUTPUT_ATTR_1, OUTPUT_ATTR_2, ..., OUTPUT_ATTR_N
+```
+
+Every rule entry consists of input and output attributes separated by the arrow symbol - `->`. 
+Input attributes are used to match the rule entry with the SKR request and to modify [search labels](#search-labels). Output attributes are only used to modify the [search labels](#search-labels).
+In its minimal form each rule consists of a `PLAN` that the rule applies to.
+In its extended form a rule entry contains lists of input attributes and its values passed in the form of `ATTR=VAL` pairs in parantheses.
+
+> ![NOTE]
+> The following rule entries are considered valid and the same:
+> * `aws`
+> * `aws()`
+
+Output attributes does not support values.
+Description of when the rule is triggered can be found in the [Rule Evaluation](#evaluation) section.
+Rule Attributes are described in [Rule Attributes](#rule-attributes) section. 
+
+Currently, possible OUTPUT_ATTR_x attributes values are `S` or `E`.
+
+```
+hap: 
+  rule: 
+  - azure(INPUT_ATTR_1=VAL_1,INPUT_ATTR_2=VAL_2,...,INPUT_ATTR_N) -> OUTPUT_ATTR_1=VAL_1,OUTPUT_ATTR_2=VAL_2,...,OUTPUT_ATTR_N
+```
+
+The input attribute types include **platformRegion** (`PR`), **clusterRegion** (`CR`). The output attribute types include **shared** (`S`) and **euAccess** (`E`). 
+Each attribute can be used at most once in a single rule entry.
+**shared** and **euAccess** attributes are used only to apply their labels modifications if rest of matched attributes are equal to SKR's values. 
+
+Rule Entry results include `S` and `E`. They are described in [Rule Results](#rule-results) section.
```

Comment:
link doesnt work

Reply:
Removed. Weird old sentence that is redundant because of describing output attributes in the paragraph above.